mantis-server/mantis-server-worker-client/src/main/java/io/mantisrx/server/worker/client/SseWorkerConnectionFunction.java

```diff
@@ -58,25 +57,23 @@ public void call(Throwable throwable) {
         // Use single netty thread
         NettyUtils.setNettyThreads();
 
-        metricGroupId = new MetricGroupId(DROP_OPERATOR_INCOMING_METRIC_GROUP + "_SseWorkerMetricsConnectionFunction_withBuffer");
-        metricsSet.add(metricGroupId);
+        metricGroup = (DROP_OPERATOR_INCOMING_METRIC_GROUP + "_SseWorkerMetricsConnectionFunction_withBuffer");
+        metricsSet.add(metricGroup);
         logger.info("SETTING UP METRICS PRINTER THREAD");
         new ScheduledThreadPoolExecutor(1).scheduleWithFixedDelay(new Runnable() {
             @Override
             public void run() {
                 try {
-                    Set<MetricGroupId> metricGroups = new HashSet<>(metricsSet);
+                    Set<String> metricGroups = new HashSet<>(metricsSet);
                     if (!metricGroups.isEmpty()) {
-                        for (MetricGroupId metricGroupId : metricGroups) {
-                            final Metrics metric = MetricsRegistry.getInstance().getMetric(metricGroupId);
-                            if (metric != null) {
-                                final Counter onNext = metric.getCounter("" + DropOperator.Counters.onNext);
-                                final Counter onError = metric.getCounter("" + DropOperator.Counters.onError);
-                                final Counter onComplete = metric.getCounter("" + DropOperator.Counters.onComplete);
-                                final Counter dropped = metric.getCounter("" + DropOperator.Counters.dropped);
-
-                                logger.info(metricGroupId.id() + ": onNext=" + onNext.value() + ", onError=" + onError.value() +
-                                                ", onComplete=" + onComplete.value() + ", dropped=" + dropped.value()
+                        for (String metricGroup : metricGroups) {
+                            if (metricGroup != null) {
+                                final Counter onNext = meterRegistry.counter(metricGroup + "_DropOperator_" + ("" + DropOperator.Counters.onNext));
```

Comment:
this will also be using `Search.in(meterRegistry).name(n -> n.startsWith(metricGroup) && n.endsWith(DropOperator.Counters.onNext.toString())).counter()`